### PR TITLE
Joukowski Camber-Corrected Cp: geometry-aware panel physics

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current Panel Cp (PR #2319) uses a flat-plate approximation: Cp = -2sin(α)/√(x(1-x)). This is geometry-agnostic — every airfoil at the same AoA gets identical Cp. Adding **thin-airfoil theory camber corrections** (A0, A1 Fourier coefficients from the airfoil camber line) makes the Cp feature geometry-aware, capturing the actual suction peak location and magnitude for each specific airfoil profile.

For tandem cases, the aft foil sees a different effective AoA due to wake interference from the fore foil. A camber-corrected Cp would capture the geometry×flow interaction more precisely. This is the natural next step after PR #2319 succeeded.

**Reference:** Thin airfoil theory (Kutta-Joukowski, standard aerodynamics). The Fourier cosine series A0..An are deterministic from the camber line slope. NeuralFoil (arXiv:2503.16323, Mar 2025) discusses why thin-airfoil Fourier coefficients outperform flat-plate approximations.

### Changes to `cfd_tandemfoil/train.py`

1. **Add flag:**
   ```python
   parser.add_argument('--cp_joukowski', action='store_true', help='Camber-corrected Cp via thin-airfoil theory')
   ```

2. **Compute camber-corrected Cp.** For NACA 4-digit airfoils (m=max camber, p=camber position), the camber slope dy_c/dx is:
   ```python
   def naca_camber_corrected_cp(x, alpha, m, p, n_terms=4):
       """
       x: arc-length parameter in [0,1] (chord fraction)
       alpha: AoA in radians
       m: max camber (e.g. 0.02 for NACA 2412)
       p: camber position (e.g. 0.4 for NACA 2412)
       """
       # Transform to cosine spacing
       theta = torch.arccos(1 - 2 * x.clamp(0, 1))  # [N]
       
       # Flat plate term
       A0 = alpha  # A0 = alpha - (1/pi)*integral of camber slope
       
       # Camber slope (NACA 4-digit)
       # dyc_dx for x < p: 2m/p^2 * (p - x)
       # dyc_dx for x >= p: 2m/(1-p)^2 * (p - x)
       dyc = torch.where(x < p,
           2*m/p**2 * (p - x),
           2*m/(1-p)**2 * (p - x))
       
       # A0 correction: subtract mean camber effect
       # A0 = alpha - (1/pi) * integral(dyc * d_theta/dx)
       # Approximation via trapezoidal rule
       A0_corr = alpha - (dyc.mean() * 1.0)  # simplified
       
       # Corrected Cp (thin-airfoil theory first order)
       Cp_corrected = -2 * (A0_corr / torch.sqrt(x.clamp(1e-6, 1) * (1 - x.clamp(0, 0.9999))))
       
       return Cp_corrected
   ```

   If the NACA parameters (m, p) are available in the dataset features, use them. If not, use a reasonable default (m=0.02, p=0.4 for a NACA 2412-like profile) as a first-order correction.

3. **Simpler alternative (if NACA params not available):** Just use the flat-plate Cp but modulate it by the DSDF surface curvature as a proxy for camber. Apply a scale factor `(1 + curvature_weight * local_curvature)` where curvature is estimated from adjacent node distances. This is less principled but still geometry-aware.

4. **Keep all existing flags.** Add `--cp_joukowski` as an ADDITIVE correction on top of `--cp_panel`. Scale by `--cp_panel_scale 0.1` as before.

5. **Run 2 seeds:**
   ```bash
   CUDA_VISIBLE_DEVICES=0 python train.py \
     --agent edward --wandb_name "edward/joukowski-s42" --wandb_group joukowski-camber-cp \
     --seed 42 --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
     --slice_num 96 --cosine_T_max 150 --pcgrad_3way \
     --pressure_first --pressure_deep --residual_prediction --surface_refine \
     --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 \
     --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 \
     --wake_angle_feature \
     --cp_joukowski

   CUDA_VISIBLE_DEVICES=1 python train.py \
     --agent edward --wandb_name "edward/joukowski-s73" --wandb_group joukowski-camber-cp \
     --seed 73 [same flags]
   ```

   Note: **include `--wake_angle_feature`** since that was just merged as the new baseline.

## NEW Baseline (PR #2350, just merged)
| Metric | 2-seed avg | Target to beat |
|--------|-----------|----------------|
| p_in | 11.90 | < 11.90 |
| **p_oodc** | **7.35** | < 7.35 |
| **p_tan** | **27.20** | < 27.20 |
| **p_re** | **6.40** | < 6.40 |

Reproduce: `cd cfd_tandemfoil && python train.py --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 --slice_num 96 --cosine_T_max 150 --pcgrad_3way --pressure_first --pressure_deep --residual_prediction --surface_refine --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 --wake_angle_feature`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT edward:

## Results

### Metrics Summary

| Metric | S42 (sys3eddk) | S73 (mnsm24ft) | 2-seed avg | Baseline (PR #2350) | Δ |
|--------|----------------|----------------|-----------|---------------------|---|
| **p_in** | 12.34 | 12.00 | **12.17** | 11.90 | +2.3% ❌ |
| **p_tan** | 27.2 | 27.3 | **27.25** | 27.20 | +0.2% ❌ |
| **p_oodc** | 7.5 | 7.2 | **7.35** | 7.35 | 0.0% — |
| **p_re** | 6.5 | 6.5 | **6.50** | 6.40 | +1.6% ❌ |

Best checkpoints: S42 epoch 160, S73 epoch 161. Peak memory: 40.6 GB / 40.4 GB.

### Command used

```bash
CUDA_VISIBLE_DEVICES=0 python train.py \
  --agent edward --wandb_name "edward/joukowski-s42" --wandb_group joukowski-camber-cp \
  --seed 42 --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --slice_num 96 --cosine_T_max 150 --pcgrad_3way \
  --pressure_first --pressure_deep --residual_prediction --surface_refine \
  --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 \
  --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 \
  --wake_angle_feature \
  --cp_joukowski
```

### Implementation

The Joukowski correction computes A0 = α - (1/π)∫₀^π (dyc/dx)dθ for each foil using the NACA 4-digit camber line formula. The raw NACA params (stored as digit/9 by parse_naca) are converted to physical values: m = stored*0.09 (max camber fraction), p = stored*0.9 (camber position). The correction is zero for symmetric foils (m=0). Fore and aft foils get separate A0 values. The effective AoA A0 replaces raw alpha in the existing Cp formula.

For NACA 2412: A0 correction ≈ 0.26° (small but non-zero), consistent with thin-airfoil theory.

### What happened

The camber correction made things slightly worse across the board. The 2-seed average for p_in went from 11.90 to 12.17 (+2.3%), and p_re also regressed. p_oodc was flat.

Analysis of why this likely failed:
1. **The A0 correction is very small** (~0.26° for NACA 2412) relative to the AoA range in the dataset (±15° or more). The signal-to-noise ratio is low.
2. **The correction modifies a feature that already works** — the flat-plate Cp was a useful signal. The small perturbation added noise rather than information.
3. **The model learns residuals anyway** — the Transolver backbone can already implicitly learn geometry-dependent corrections from the NACA features in the input (x[:, :, 15:17]). The A0 correction may be redundant.
4. **Tandem cases**: The aft foil's effective AoA is dominated by wake interference, not just geometric camber. Using the geometric A0 for the aft foil ignores wake-induced AoA changes.

The NACA geometry is already in the input features (channels 15-17), so the model can learn the camber effect directly. Making the Cp feature geometry-aware via A0 adds minimal new information.

### Suggested follow-ups

- **Higher-order correction**: Include the A1 term (sin(θ) component), which contributes ~10× more than A0 to the actual CL. This might show a larger signal.
- **Separate feature channel**: Instead of replacing the Cp feature, add A0 as a new per-node channel (geometry-aware correction term), letting the model decide how to use it.
- **Wake-induced AoA for aft foil**: Use the fore-foil wake model to estimate effective AoA for aft-foil nodes, rather than geometric camber only.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2356 because ADVISOR: Closing — all metrics regressed or flat (p_in +2.3%, p_re +1.6%, p_oodc flat, p_tan +0.2%). Good analysis on why: the A0 camber correction is ~0.26° — far too small relative to the AoA range to be useful. The model already learns camber effects from the NACA geometry channels (15-17). This confirms our broader finding: incremental refinements to the existing Cp physics feature are exhausted. The flat-plate approximation scaled by 0.1 is sufficient — the backbone handles the rest.

Edward: your next assignment will be a fundamentally different experiment (Round 40 bold direction). Stand by..
